torchaudio/datasets/utils.py

```diff
@@ -221,3 +223,26 @@ def __getitem__(self, n):
 
     def __len__(self):
         return len(self.dataset)
+
+
+class BackgroundGenerator(threading.Thread):
+    def __init__(self, generator, queue_length=1):
+        threading.Thread.__init__(self)
+        self.queue = Queue(queue_length)
+        self.generator = generator
+        self.daemon = True
+        self.start()
+
+    def run(self):
+        for item in self.generator:
+            self.queue.put(item)
+        self.queue.put(None)
```

Comment:
An iterator might very well return None (such as indicating an empty datapoint). We should use a custom poison bill, something like a very specific class

Reply:
> 2. Wrap this into a function such as `bg_iterator(iterable, queue_size=1)` to make it more consistent with download, iterator etc.

I've just renamed the class. What would wrapping the class in a function add? The notation is already that of a function:
```
for i in bg_iterator(range(10)):
  print(i)
```

Reply:
Created a class _End to be the end of queue, and checking for this.